demos/classification_benchmark_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "custom_kernels.hpp"
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <stdexcept>
+#include <utility>
+
+#include <ie_core.hpp>
+#include <ie_data.h>
+#include <ie_layouts.h>
+#include <opencv2/gapi/core.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+#include <opencv2/gapi/gscalar.hpp>
+#include <opencv2/gapi/imgproc.hpp>
+#include <opencv2/gapi/infer.hpp>
+#include <opencv2/gapi/operators.hpp>
+#include <opencv2/gapi/own/assert.hpp>
+#include <opencv2/imgproc.hpp>
+
+
+void IndexScore::getScoredLabels(const std::vector<std::string> &in_labes,
+                                 LabelsStorage &out_scored_labels_to_append) const {
+    uint64_t labels_offset_correction = 0;
+    if (in_labes.size() + 1  == classification_indices_scale) {
+        // Equal to inserting 'other' label as first.
+        labels_offset_correction = -1;
+    } else if (classification_indices_scale != in_labes.size()) {
+        throw std::logic_error("Model's number of classes and parsed labels must match (" +
+                               std::to_string(classification_indices_scale) + " and " +
+                               std::to_string(in_labes.size()) + ')');
+    }
+
+    // fill starting from max confidence
+    for (auto conf_index_it = max_confidence_with_indices.rbegin();
+         conf_index_it != max_confidence_with_indices.rend();
+         ++conf_index_it) {
+        size_t recalculated_label_id = *conf_index_it->second + labels_offset_correction;
+        try {
+            out_scored_labels_to_append.emplace_back(conf_index_it->first,
+                                                     *conf_index_it->second,
+                                                     recalculated_label_id != 0 ? in_labes.at(recalculated_label_id) : "others");
+        } catch (const std::out_of_range& ex) {
+            throw std::out_of_range(std::string("Provided labels file doesn't contain classified label index\nException: ") + ex.what());
+        }
+    }
+}
+
+IndexScore IndexScore::create_from_array(const float *out_blob_data_ptr, size_t out_blob_element_count,
```

Comment:
Wow, I expected to see it as small function that just use `sort` :)
We have only `1000` values there...

Reply:
please check the previous comments from @Wovchena about the discussion of using `nth_element`. All of them applied in case of `std::sort` as well

Reply:
if comments were lost: 
sort as well as other std algorithms  require additional memory allocations for both vertices & indices either; copy vertices + fill (by iota) indices; + custom modifications for keep stable indices in parallel. On every frame
So it makes no sense to bring up such a premature pessimization if  you has better solution already